src/fallback.ts

```diff
@@ -313,30 +325,90 @@ export class GrpcClient {
         const protoServiceName = protoNamespaces.join('.');
         const rpcName = method.name;
 
-        const url = `${grpcFallbackProtocol}://${servicePath}:${servicePort}/$rpc/${protoServiceName}/${rpcName}`;
+        let url: string;
+        let data: string;
+        let httpMethod: string;
+
+        if (this.fallback === 'json') {
+          // REGAPIC: JSON over HTTP/1 with gRPC trancoding
+          headers['Content-Type'] = 'application/json';
+          const decodedRequest = method.resolvedRequestType.decode(requestData);
```

Comment:
I would be tempted to pull this logic into a helper, it's getting chunky enough that it would hep the readability.

Reply:
I agree with that. This will require more work than just copy-pasting since some of the variables don't have the proper types defined, and passing them as parameters would be ugly (unless we use `any`, which is ugly too). I will leave a TODO here for now.